platform-sdk/docs/proposals/state-lifecycle/proposal.md

```diff
@@ -0,0 +1,231 @@
+# Lifecycle management for the State
+
+## Summary
+
+This proposal describes a possible implementation of state lifecycle management that could work equally well for both the Consensus Node and the Block Node.
+
+|      Metadata      |                 Entities                 |
+|--------------------|------------------------------------------|
+| Designers          | [@imalygin](https://github.com/imalygin) |
+| Functional Impacts | Consensus Node, Block Node               |
+
+## Purpose and Context
+
+There is an initiative to create self-contained State modules (`swirlds-state-api` and `swirlds-state-impl`) that would have minimal dependencies, and specifically, they should not depend on `swirlds-platform-core`. Part of this initiative is the implementation of a state lifecycle management mechanism that could be used by both the Consensus Node and the Block Node.
+
+### Requirements
+
+- Add an interface representing state lifecycle management and its implementation.
+- Update the current State API to be coherent with this State Lifecycle API.
+- Implement this set of APIs and migrate the existing code to use it.
+
+## Design & Architecture
+
+### Reservation count mechanism
+
+This mechanism is already implemented as a part of the Merkle tree implementation. However, it's importnant to mention it here
```

Comment:
```suggestion
This mechanism is already implemented as a part of the Merkle tree implementation. However, it's important to mention it here
```

Reply:
I didn't just fixed the typo, I reworded the whole sentence.